app/feature/feature-chat/src/main/kotlin/com/hedvig/android/feature/chat/ChatViewModelNew.kt

```diff
@@ -0,0 +1,161 @@
+package com.hedvig.android.feature.chat
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.viewModelScope
+import com.hedvig.android.core.demomode.Provider
+import com.hedvig.android.feature.chat.data.ChatMessage
+import com.hedvig.android.hanalytics.featureflags.FeatureManager
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import java.io.File
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.datetime.Instant
+
+internal class ChatViewModelNew(
+  chatRepository: Provider<ChatRepositoryNew>,
+  chatClosedTracker: ChatEventStore,
+  featureManager: FeatureManager,
+) : MoleculeViewModel<ChatEventNew, ChatUiState>(
+  ChatUiState(),
+  ChatPresenter(
+    chatRepository = chatRepository,
+    chatClosedTracker = chatClosedTracker,
+    featureManager = featureManager,
+  ),
+) {
+  init {
+    emit(ChatEventNew.FetchMessages())
+    viewModelScope.launch {
+      while (true) {
+        delay(5.seconds)
+        emit(ChatEventNew.PollMessages)
+      }
+    }
+  }
+}
+
+internal class ChatPresenter(
+  private val chatRepository: Provider<ChatRepositoryNew>,
+  private val chatClosedTracker: ChatEventStore,
+  private val featureManager: FeatureManager,
+) : MoleculePresenter<ChatEventNew, ChatUiState> {
+
+  @Composable
+  override fun MoleculePresenterScope<ChatEventNew>.present(lastState: ChatUiState): ChatUiState {
+    var isLoadingChat by remember { mutableStateOf(lastState.isLoadingChat) }
+    var isLoadingNewMessages by remember { mutableStateOf(lastState.isLoadingNewMessages) }
+    var messages by remember { mutableStateOf(lastState.messages) }
+    var isSendingMessage by remember { mutableStateOf(lastState.isSendingMessage) }
+    var canFetchMoreMessages by remember { mutableStateOf(lastState.canFetchMoreMessages) }
```

Comment:
Is this to be used by the UI itself, so that it know when *not* to send new requests for new messages? Or what is the idea?